test/core/gpr/arena_test.cc

```diff
@@ -86,7 +81,7 @@ typedef struct {
 static void concurrent_test_body(void* arg) {
   concurrent_test_args* a = static_cast<concurrent_test_args*>(arg);
   gpr_event_wait(&a->ev_start, gpr_inf_future(GPR_CLOCK_REALTIME));
-  for (size_t i = 0; i < concurrent_test_iterations(); i++) {
+  for (size_t i = 0; i < 1000; i++) {
```

Comment:
I am actually not sure what would be the right constant to use here.  1000 was chosen for 32bit platforms.  Have you tried if things still pass with remote bazel at 10000 iterations? (to me it seems that the main thing that would influence memory consumption is the number of threads.

Reply:
Yes, it looks like only scaling back threads is enough to prevent the timeout (at least on my machine)